### PR TITLE
Update resource cache after update/create from Dependent Resource

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java [hunk 1]
```diff
@@ -39,4 +40,8 @@ public void stop() {
     super.stop();
     manager().stop();
   }
+
+  public void populateCacheUpdatedResource(R resource) {
+    cache.put(ResourceID.fromResource(resource), resource);
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java:45]: Based on https://github.com/fabric8io/kubernetes-client/pull/3654#issuecomment-1026838969, I'm not sure we should be modifying the informer's cache :(
Another thing is that this won't be possible in 6.0 anymore so we need to think about an alternative, in particular in the light of: https://github.com/fabric8io/kubernetes-client/pull/3616. So we might have to think about re-adding our own caching layer sooner than later…

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java:45]: Yes, the plan is to do it with the separate cache if the informers will be modified for that. Not sure if we can get away with this until that time. It's not 100% correct, but the chance to have there a race condition is very close to zero

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java:45]: An alternative is to wait for the event that the cache received the resource with the resource version.

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java:45]: That would mean not the update/delete would propagate the cache but the `getResource()` will be kinda "synchronous"

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java]: Solved it in a different way, pls see updated PR. (Will add unit tests soon)

**Comment by metacosm** [comment]
I'm thinking that this work should be integrated in the `InformerResourceCache` layer instead of a separate layer that needs to be managed by, and therefore only useful to, `KubernetesDependentResource`. This would make the functionality available even for people not using `KubernetesDependentResource` and solve the problematic informer cache update that's currently performed by `InformerResourceCache` in a neat way. What do you think?

**Comment by csviri** [comment]
The thing is that this is not needed when you work directly with resources, since when a resource is created by the client a new object is returned that users can work with further. So it us really useful for dependent resources. Since the getResource() is called eventually by other dependent resource. 

Plan is to put it there yes, after the cache layer will be separated from informers, as mentioned by @shawkins; for an upcoming version of informers. So for now I would just merge this, and create a followup issue.

Putting this updated resource now to cache of the informers has race condition issues, that are  not nicely solvable now.

**Comment by csviri** [comment]
just a remark this issue I actually plan to put into informer event source: https://github.com/java-operator-sdk/java-operator-sdk/issues/870

It's kinda similar to this. So we can revisit also there. Not sure we need this to MVP though, maybe right after?

**Comment by csviri** [comment]
Hmm, actually can take a look if you insist, now this impl looks quite different then on begging. Maybe would be able to implement the mentioned issue in a simple way.

**Comment by csviri** [comment]
this PR is replaced by: https://github.com/java-operator-sdk/java-operator-sdk/pull/965

**Pull request closed without merging**